modules/ROOT/pages/persistence-gateway.adoc

```diff
@@ -0,0 +1,122 @@
+| Maximum key length | 256 characters
+| Maximum value length | 10 MB
+| Maximum partitions per application | 10
+| Maximum keys per partition | 1000
+|===
+
+
+[Configure-Persistence-Gateway]
+== Configure Persistence Gateway
+
+To enable Persistence Gateway, you must create a Kubernetes custom resource that allows the cluster to connect to your persistence data store.
+
+=== Prerequisties
+
+Before enabling Persistence Gateway, ensure that you have the following:
+
+* A data source to contain the data stored by Persistence Gateway. This data source must be compatible with PostgresSQL 9 and above.
+* A Kubernetes secret containing the network information and credentials to connect to the data store.
+
+=== Create a Custom Resource and Secret
+
+. Create a Kubernetes custom resource for your data store.
+.. Copy the custom resource template from <<Kubernetes Custom Resource Template>> to a file called `custom-resource.yaml`.
+.. Modify the fields of the template as required for your environment.
+.. Run `kubectl apply -f custom-resource.yaml` to add the custom resource to your cluster.
+. Create a Kubernetes secret.
+.. Copy the Kubernetes secret template from <<Kubernetes Secret Template>> to a file called `kubernetes-secret.yaml`.
+.. Modify the fields of the template to include connectivity information and credentials for your installation.
+.. Run `kubectl apply -f kubernetes-secret.yaml`.
+
+=== Kubernetes Custom Resource Template
+
+Use the example template below to create a Kubernetes custom resource. This custom resource enables the Kubernetes cluster to connect to your data store.
+
+----
+apiVersion: rtf.mulesoft.com/v1
+kind: PersistenceGateway
+metadata:
+	name: persistent-gateway-config
+	namespace: rtf
+spec:
+	objectStore:
+    	backendDriver: postgresql
+    	maxBackendConnectionPool: 10
+		maxKeysPerStore: 5000
+    	maxStores: 4
+    	replicas: 1
+		resources:
+			limits:
+				cpu: 1m
+        		memory: 200Mi
+			requests:
+				cpu: 0.5m
+				memory: 100Mi
+		secretRef:
+			name: persistence-gateway-creds
+----
+
+Below are descriptions of the fields of the template you modify:
+
+[%header,cols="3*a"]
+|===
+| Field | Description | Default Value
+| `backendDriver` | Specifies the driver used by the data store. Only `postgresql` is supported. | `postgresql`
+| `maxBackendConnectionPool` | | 10
+| `maxKeysPerStore` | | 5000
+| `maxStores` | | 5000
+| `replicas` | | 5000
+| `resources: limits: cpu` | | 1m
+| `resources: limits: memory` | | 200Mi
+| `resources: requests: cpu` |  | 0.5m
+| `resources: requests: memory` | | 100Mi
+| `secretRef: name:` | Specifies the name of the persistence gateway credentials defined in the Kubernetes secret file.| `persistence-gateway-creds`
+|===
+
+=== Kubernetes Secret Template
+
+Use the example template below to create a Kubernetes secret. The Kubernetes cluster uses this secret to connect to your data store.
```

Comment:
I think we also need to explicitly call out few additional points here:

- the name of the secret can be per the customer's liking
- the data section of the secret is where the connection string would reside as the value of the key `persistence-gateway-creds:`
- the secret needs to be created in the rtf namespace
- we should also consider specifying the connection string format